collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -312,6 +407,9 @@ public void refreshView() {
 
                 switch (event) {
                     case FormEntryController.EVENT_QUESTION:
+                        if (shouldShowRepeatGroupPicker()) {
```

Comment:
What case does it cover? I mean how is it possible that we have `EVENT_QUESTION` but repeat group should be displayed?

Reply:
This switch statement is part of a loop that goes through the entire form to render every item in the hierarchy. It skips items outside the current repeat group, but otherwise, it will render every question and nested repeat instance.

This code prevents questions from being rendered inside the form hierarchy picker, ensuring that only `CHILD` elements (nested repeat instances) are displayed.